### DISCUSSION TITLE
Not able to run exec command to get into velero pod

### DISCUSSION THREAD

**Discussion by vipin5554** [vmware-tanzu/velero#4787] [Community support Q&A]
I am trying to login velero pod running in Azure Kubernetes service with below command

kubectl exec -it velero-fdf5458b6-slgjz -n velero -- sh
kubectl exec -it velero-fdf5458b6-slgjz -n velero -- /bin/sh
kubectl exec -it velero-fdf5458b6-slgjz -n velero -- /bin/bash

Run all command to get into pod but getting error

Defaulted container "velero" out of: velero, velero-velero-plugin-for-microsoft-azure (init)
error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "7c19c59856b6b5c16a3397dd4178d6257a93b7147ad7805581f2153a94c849e5": OCI runtime exec failed: exec failed: container_linux.go:380: starting container process caused: exec: "sh": executable file not found in $PATH: unknown

Please suggest.

**Comment by venkat-jaligama** [comment]
I am getting same error

**Comment by albertchua** [comment]
I'm also getting the error using v1.8.1 in Rancher.

**Reply by loctran1991** [reply]
I have same issue, So I tried this command and it works for me
kubectl exec -it velero-fdf5458b6-slgjz -n velero -- /velero schedule get
kubectl exec -it velero-fdf5458b6-slgjz -n velero -- /velero backup get

**Reply by minhtrung071190** [reply]
It works for me. Thank you

**Comment by adamrushuk** [comment]
I got around this by using an older version (`v1.6.3`)

**Comment by VickyWinner** [comment]
I am below error.  Tried all combinations.
```
kc exec pod/velero-6d49f99b4f-z4vgf -c velero -- /bin/sh
error: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec
 "c39630fe55fdb9320b75e52200e3a6e9dc639d1d829f9a7dd31004471fda24af": OCI runtime exec failed: 
exec failed: container_linux.go:380: starting container process caused: exec: "//bin/bash": stat //bin/bash: no such file
or directory: unknown

**Comment by adamrushuk** [comment]
The ` OCI runtime exec failed` error occurs due to velero using distroless images that have no shell, like bash. 

Consider using `kubectl debug` to start an ephemeral container: https://kubernetes.io/docs/tasks/debug/debug-application/debug-running-pod/#ephemeral-container
